hypothesis-python/src/hypothesis/database.py

```diff
@@ -671,3 +673,75 @@ def move(self, src: bytes, dest: bytes, value: bytes) -> None:
 
     def delete(self, key: bytes, value: bytes) -> None:
         raise RuntimeError(self._read_only_message)
+
+
+def ir_to_bytes(ir: Iterable[IRType], /) -> bytes:
```

Comment:
This implementation is lifted from [your branch](https://github.com/HypothesisWorks/hypothesis/compare/master...Zac-HD:hypothesis:ir-serializer), with two fixes/changes:

- adjust int for negative values
- surrogatepass instead of surrogateescape (I don't recall my at-the-time justification of this, but if you run your test case for long enough you will get an error with surrogateescape)